### ISSUE TITLE
ThreadStartString_01 fails on non-windows

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/coreclr#3627]
src/tests/baseservices/threading/paramthreadstart/ThreadStartString_01

**Comment by sergiy-k** [comment]
This test does not fail when executed directly from the command line. The issue seems to be caused by the auto generated shell file.

**Comment by jashook** [comment]
Related to #18381?

**Comment by jashook** [comment]
```
baseservices_threading._paramthreadstart_ThreadStartBool_1_ThreadStartBool_1_._paramthreadstart_ThreadStartBool_1_ThreadStartBool_1_sh [FAIL] (TaskId:6)
                           /bin/sh: /users/jashoo/projects/coreclr/tests/bin/tests/OSX.x64.Debug/baseservices/threading/paramthreadstart/ThreadStartBool/ThreadStartBool.sh: No such file or directory (TaskId:6)
                           /n/nReturn code:      1/nRaw output file:      /Users/jashoo/projects/coreclr/bin/tests/OSX.x64.Debug/baseservices/threading/Reports/baseservices.threading/paramthreadstart/ThreadStartBool_1/ThreadStartBool_1.output.txt/nRaw output:/nBEGIN EXECUTION (TaskId:6)
                           export CDPATH=/Users/jashoo/projects/coreclr/bin/tests/OSX.x64.Debug/baseservices/threading/paramthreadstart/ThreadStartBool_1 (TaskId:6)
                           /bin/sh -c /users/jashoo/projects/coreclr/tests/bin/tests/OSX.x64.Debug/baseservices/threading/paramthreadstart/ThreadStartBool/ThreadStartBool.sh (TaskId:6)
                           Expected: 0 (TaskId:6)
                           Actual: 127 (TaskId:6)
                           END EXECUTION - FAILED (TaskId:6)
                           Test Harness Exitcode is : 1 (TaskId:6)
                           /nTo run the test:/n> set CORE_ROOT=/Users/jashoo/projects/coreclr/bin/tests/OSX.x64.Debug/Tests/Core_Root/n> /Users/jashoo/projects/coreclr/bin/tests/OSX.x64.Debug/baseservices/threading/paramthreadstart/ThreadStartBool_1/ThreadStartBool_1.sh/n (TaskId:6)
                           Expected: True (TaskId:6)
                           Actual:   False (TaskId:6)
                           Stack Trace: (TaskId:6)
                             /Users/jashoo/projects/coreclr/bin/tests/OSX.x64.Debug/TestWrappers/baseservices.threading/baseservices.threading.XUnitWrapper.cs(249,0): at baseservices_threading._paramthreadstart_ThreadStartBool_1_ThreadStartBool_1_._paramthreadstart_ThreadStartBool_1_ThreadStartBool_1_sh() (TaskId:6)
```

**Comment by jashook** [comment]
Closing as a dup of #18381

**Comment by sandreenko** [comment]
The test continues to fail after fix the fix for #18381.

**Comment by echesakovMSFT** [comment]
Seems that this is not a test issue, it's a test runner issue. Will submit fix shortly along with a description of the problem
